acquia_lift.admin.wizard.inc

```diff
@@ -1030,6 +1030,7 @@ function acquia_lift_personalize_campaign_wizard_submit(&$form, &$form_state) {
 
   // Run the base submit form handling.
   $agent_data = personalize_campaign_wizard_submit_base($form, $form_state);
+  $agent_data = personalize_agent_load($agent_data->machine_name);
   $agent_instance = personalize_agent_load_agent($agent_data->machine_name);
```

Comment:
Exactly.... instead of just reading from the base form - since we have extra stuff now it's like the callback should instead load the agent and then update the label based on the ever-present agent_basic_info fields rather than loading directly from the agent_basic_info fields and assuming that they encompass everything.